openedx_tagging/core/tagging/import_export/import_plan.py

```diff
@@ -136,7 +136,7 @@ def generate_actions(
 
         if replace:
             tags_for_delete = {
-                tag.external_id: tag for tag in self.taxonomy.tag_set.all()
+                tag.external_id or tag.id: tag for tag in self.taxonomy.tag_set.all()
```

Comment:
Because of this change, I think we also need to change the code below:
https://github.com/openedx/openedx-learning/blob/fb8d16112fa4705fda8f8ac44cd3401cfb08619e/openedx_tagging/core/tagging/import_export/import_plan.py#L96

Context: this is to prevent updating a tag that could already be deleted

The following test (to be put in `import_export/test_api.pu` reproduce the issue:
```
    def test_import_removing_with_childs_no_external_id(self) -> None:
        """
        Test import need to remove childs with parents that will also be removed,
        using tags without external_id
        """
        new_taxonomy = Taxonomy(name="New taxonomy")
        new_taxonomy.save()
        level2 = Tag.objects.create(
            id=1000,
            value="Tag 2",
            taxonomy=new_taxonomy,
        )
        level1 = Tag.objects.create(
            id=1001,
            value="Tag 1",
            taxonomy=new_taxonomy,
        )
        level3 = Tag.objects.create(
            id=1002,
            value="Tag 3",
            taxonomy=new_taxonomy,
        )
        level2.parent = level1
        level2.save()

        level3.parent = level3
        level3.save()

        # Import with empty tags, to remove all tags
        importFile = BytesIO(json.dumps({"tags": []}).encode())
        result, _tasks, _plan =  import_export_api.import_tags(
            new_taxonomy,
            importFile,
            ParserFormat.JSON,
            replace=True,
        )
        breakpoint()
        assert result
```

Reply:
Good catch! Thanks!

Reply:
@rpenido  Updated here: https://github.com/openedx/openedx-learning/pull/135/commits/b1c4cc902182b48bf902241f69148693d81422d2